### ISSUE TITLE
Parameter for role(s) when using customer endpoints

### ISSUE THREAD

**Issue by bryceadams** [woothemes/woocommerce#7289]
At the moment the customer endpoint only looks up users with the `customer` role. There is no way to override this with a custom param. So we will add in a check for a new custom param for `role`, but another issue is that we need to support an array of roles. Eg. someone wants users that are either `customer` or `subscriber` (cc @thenbrent). 

The only way at the moment in WP to look up users in 2 roles is to do this (sort of bad) query:

```
global $wpdb;
$blog_id = get_current_blog_id();

$user_query = new WP_User_Query( array(
    'meta_query' => array(
        'relation' => 'OR',
        array(
            'key' => $wpdb->get_blog_prefix( $blog_id ) . 'capabilities',
            'value' => 'role_one',
            'compare' => 'like'
        ),
        array(
            'key' => $wpdb->get_blog_prefix( $blog_id ) . 'capabilities',
            'value' => 'role_two',
            'compare' => 'like'
        )
    )
) );
```

So we will do this. People can then pass in a array for the role parameter and it will loop through the roles.

**Comment by bryceadams** [comment]
As discussed with @mikejolley, we'd prefer not to use a 'hack' to query multiple roles, so this will allow just a single role to be overridden.

**Comment by claudiosmweb** [comment]
Closed in favor to #7291
